loleaflet/src/core/Socket.js

```diff
@@ -1130,20 +1129,25 @@ app.definitions.Socket = L.Class.extend({
 				this._map.options.docParams['reuse_cookies'] = reuseCookies;
 			}
 
-			// setup for loading the new document, and trigger the load
-			var docUrl = url.split('?')[0];
-			this._map.options.doc = docUrl;
-			this._map.options.previousWopiSrc = this._map.options.wopiSrc; // After save-as op, we may connect to another server, then code will think that server has restarted. In this case, we don't want to reload the page (detect the file name is different).
-			this._map.options.wopiSrc = encodeURIComponent(docUrl);
-
 			// if this is save-as, we need to load the document with edit permission
 			// otherwise the user has to close the doc then re-open it again
 			// in order to be able to edit.
 			if (textMsg.startsWith('saveas:'))
 				this._map.options.permission = 'edit';
-			this._map.loadDocument();
-			this._map.sendInitUNOCommands();
 
+			// setup for loading the new document, and trigger the load
+			var docUrl = url.split('?')[0];
+			this._map.options.doc = docUrl;
+			this._map.options.previousWopiSrc = this._map.options.wopiSrc; // After save-as op, we may connect to another server, then code will think that server has restarted. In this case, we don't want to reload the page (detect the file name is different).
+			var encodedURL = encodeURIComponent(docUrl);
+			var reload = this._map.options.wopiSrc != encodedURL;
+			this._map.options.wopiSrc = encodedURL;
+
+			if (reload) {
```

Comment:
Minor point: since you have previousWopiSrc, you could avoid the `reload` variable and replace these 4 lines with:
```
			this._map.options.wopiSrc = encodedURL;
			if (this._map.options.previousWopiSrc != encodedURL) {
```